### PR TITLE
#167313388 configure eslint 

### PR DESCRIPTION
**Pull request by salviosage**
#### What does this PR do?

- have Eslint configured in a codebase

#### Description of Task to be completed?

- add Eslint dev dependencies 

- creating .eslintrc file 

- configure eslint  to extends Airbnb-base and add other configurations as stated on [pt story](https://www.pivotaltracker.com/story/show/167313388)

#### How should this be manually tested?

- after cloning a repo  cd to your repo code base and type the command `npm install ` 

- after then you are ready to test 
#### What are the relevant pivotal tracker stories?
[#167313388](https://www.pivotaltracker.com/story/show/167313388)
#### Screenshots (if appropriate)
image
image

#### Questions:

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+  "one-var": 0,
+  "one-var-declaration-per-line": 0,
+  "new-cap": 0,
+  "consistent-return": 0,
+  "no-param-reassign": 0,
+  "comma-dangle": 0,
+  "curly": ["error", "multi-line"],
+  "import/no-unresolved": [2, { commonjs: true }],
+  "no-shadow": ["error", { "allow": ["req", "res", "err"] }],
+  "valid-jsdoc": ["error", {
```
Comment (nshutijonathan) [inline .eslintrc.js:19]: They are some rules that we have to sync and explain for us what do they do

**Review by Mnickii — APPROVED** [review]
LGTM

**Review by nshutijonathan — APPROVED** [review]
It looks good to me

**Review by minega25 — APPROVED** [review]

**Review by nkalyesubula — CHANGES_REQUESTED** [review]
Could you please transform the es5 code in .eslintrc.js to es6 code

File: .eslintrc.js [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+  }],
+  "require-jsdoc": ["error", {
+      "require": {
+          "FunctionDeclaration": true,
+          "MethodDefinition": true,
+          "ClassDeclaration": true
+      }
+  }]
+}
+};
```
Comment (Mnickii) [inline .eslintrc.js:33]: Add a new line here

**Comment by Mnickii** [comment]
@nkalyesubula I don't think that's absolutely necessary as it's a config file.

**Review by nkalyesubula — APPROVED** [review]

**Comment by Anguandia** [comment]
Looks good in to me, good work!

**Review by habinezadalvan — APPROVED** [review]

File: .eslintrc.js (hunk 1 above)
Comment (salviosage) [reply .eslintrc.js:19]: yeah sure

**Comment by Anguandia** [comment]
Looks fine, please go ahead

**Pull request merged**